packages/react-experiments/src/components/SelectedItemsList/SelectedPeopleList/Items/SelectedPersona.tsx

```diff
@@ -60,97 +60,73 @@ const SelectedPersonaInner = React.memo(
-          updateDropState: _updateDroppingState,
-        };
-
-        const events = new EventGroup(this);
-
-        const subscription = dragDropHelper?.subscribe(rootRef.current as HTMLElement, events, dragDropOptions);
-
-        return () => {
-          subscription?.dispose();
-          events.dispose();
-        };
-      },
-      // eslint-disable-next-line react-hooks/exhaustive-deps -- this is the only dependency which matters
-      [dragDropHelper],
-    );
-
-    const isDraggable = React.useMemo(
-      () => (dragDropEvents && dragDropEvents.canDrag ? !!dragDropEvents.canDrag!() : undefined),
-      [dragDropEvents],
-    );
-
-    const droppingClassName = React.useMemo(
-      () => (isDropping ? droppingClassNames || DEFAULT_DROPPING_CSS_CLASS : ''),
-      [isDropping, droppingClassNames],
-    );
-
-    const onExpandClicked = React.useCallback(
-      ev => {
-        ev.stopPropagation();
-        ev.preventDefault();
-        if (onItemChange && getExpandedItems) {
-          getExpandedItems(item)
-            .then(value => {
-              onItemChange(value, index);
-            })
-            .catch(error => {
-              // No op
-            });
+        } else if (dragDropEvents!.onDragEnter) {
+          setDroppingClassNames(dragDropEvents!.onDragEnter(item, event));
         }
-      },
-      [onItemChange, getExpandedItems, item, index],
-    );
-
-    const onRemoveClicked = React.useCallback(
-      ev => {
-        ev.stopPropagation();
-        ev.preventDefault();
-        onRemoveItem && onRemoveItem();
-      },
-      [onRemoveItem],
-    );
-
-    const itemSize = React.useMemo(() => item?.size || DEFAULT_PERSONA_SIZE, [item]);
 
-    const buttonSize = React.useMemo(
-      () => (itemSize === PersonaSize.size8 ? 8 : itemSize === PersonaSize.size24 ? 24 : 32),
-      [itemSize],
-    );
-
-    const classNames: IProcessedStyleSet<ISelectedPersonaStyles> = React.useMemo(
-      () =>
-        getClassNames(styles, {
-          isSelected: selected || false,
-          isValid: isValid ? isValid(item) : true,
-          theme: theme!,
-          droppingClassName,
-          buttonSize,
-        }),
-      [selected, isValid, theme, buttonSize, item, styles, droppingClassName],
-    );
+        if (isDropping !== newValue) {
+          setIsDropping(newValue);
+        }
+      };
+
+      const dragDropOptions: IDragDropOptions = {
+        eventMap: eventsToRegister,
+        selectionIndex: index,
+        context: { data: item, index: index },
+        ...dragDropEvents,
+        updateDropState: _updateDroppingState,
+      };
+
+      const events = new EventGroup(this);
+
+      const subscription = dragDropHelper?.subscribe(rootRef.current as HTMLElement, events, dragDropOptions);
+
+      return () => {
+        subscription?.dispose();
+        events.dispose();
+      };
+    }, [dragDropHelper, dragDropEvents, index, isDropping, eventsToRegister, item]);
```

Comment:
Is there a reason to add all the other dependencies here instead of just keeping it at dragdrophelper as previously?